avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -274,14 +274,23 @@ public Object fromConnectData(Schema schema, Object value) {
    * @param logicalValue the Connect data to convert, which may be a value for a logical type
    * @param requireContainer if true, wrap primitives, maps, and arrays in a NonRecordContainer
    *                         before returning them
+   * @param requireSchemalessContainerNull if true, use a container representation of null because
+   *                                       this is part of struct/array/map and we cannot
+   *                                       represent nulls as true null because Anything cannot
+   *                                       be a union type; otherwise, this is a top-level value
+   *                                       and can return null
    * @return the converted data
    */
   private static Object fromConnectData(Schema schema, org.apache.avro.Schema avroSchema,
-                                       Object logicalValue, boolean requireContainer) {
+                                       Object logicalValue, boolean requireContainer, boolean
+                                            requireSchemalessContainerNull) {
```

Comment:
If I understand correctly, this will always be true for map/array/struct, and always false otherwise? 

I'm wondering if we can get rid of the parameter and just use schema.type() to have this logic in one place?